### PR TITLE
Upgrade to torchaudio v0.3.0 FastAI 1.0.59 and other enhancements

### PR DESCRIPTION
**Pull request by filipmu**
Includes:

- Upgrade to torchaudio v0.3.0 covering issue: [https://github.com/mogwai/fastai_audio/issues/37#]
- Added .from_df option for loading data.
- Fixed issue with ability to handle multicategory problems
- Updated the impacted Tutorial notebooks and Getting Started notebook.
- Identified conda sources for relevant dependencies - updated the Readme.md

### CODE REVIEW

**Comment by rbracco** [comment]
Thank you for these much needed changes @filipmu! I have a few pressing things early this week, but I plan to review the PR and hopefully merge by Thursday at the latest. 

We are accepting contributors both for this version, and our repo for fastai v2 audio https://github.com/rbracco/fastai_dev. Let me know if you'd like to be added to the Telegram chat, and thanks for taking the initiative! Cheers.

File: README.md [hunk 1]
```diff
@@ -2,14 +2,21 @@
...
+```
+
+
 # Installation
 To install, run the following commands in a shell. 
 
 ```
 git clone https://github.com/mogwai/fastai_audio
 cd fastai_audio
-sudo chmod +x install.sh
```
Comment (rbracco) [inline README.md:11]: Why was the install script replaced with conda installs?

File: audio/data.py [hunk 2]
```diff
@@ -339,12 +338,13 @@ def pad_func(sig, sr):
...
         if self.config.mfcc: 
-            mel = MFCC(sr=item.sr, n_mfcc=self.config.sg_cfg.n_mfcc, melkwargs=self.config.sg_cfg.mel_args())(item.sig)
+            mel = MFCC(sample_rate=item.sr, n_mfcc=self.config.sg_cfg.n_mfcc, melkwargs=self.config.sg_cfg.mel_args())(item.sig)
         else:
             mel = MelSpectrogram(**(self.config.sg_cfg.mel_args()))(item.sig)
             if self.config.sg_cfg.to_db_scale: 
-                mel = SpectrogramToDB(top_db=self.config.sg_cfg.top_db)(mel)
-        mel = mel.permute(0, 2, 1)
+                mel = AmplitudeToDB(top_db=self.config.sg_cfg.top_db)(mel)
+        #mel = mel.permute(0, 2, 1)
```
Comment (rbracco) [inline audio/data.py]: Remove commented code

File: audio/transform.py [hunk 3]
```diff
@@ -240,11 +240,13 @@ def tfm_down_and_up(ai:AudioItem, sr_divisor=2, **kwargs)->AudioItem:
...
     return AudioItem(sig,ai.sr)
 
-def tfm_pad_to_max(ai:AudioItem, mx=1000):
-    """Pad tensor with zeros (silence) until it reaches length `mx`"""
-    copy = ai.sig.clone()
-    padded = torchaudio.transforms.PadTrim(max_len=mx)(copy[None,:]).squeeze()
-    return AudioItem(padded, ai.sr)
+# Remove since torch audio does not support PadTrim any more since v0.3
+# Also, this does not seem to be used in any other code, tutorials, getting started notebooks
+#def tfm_pad_to_max(ai:AudioItem, mx=1000):
```
Comment (rbracco) [inline audio/transform.py]: remove commented code

File: audio/data.py [hunk 4]
```diff
@@ -398,10 +398,25 @@ def _plot_lengths(self, lens, prec, figsize):
...
-            extensions = AUDIO_EXTENSIONS
-        return cls(get_files(path, extensions, recurse), path, **kwargs)
+    def from_folder(cls, path:PathOrStr='.', extensions:Collection[str]=None, **kwargs)->ItemList:
+        "Get the list of files in `path` that have an audio suffix. `recurse` determines if we search subfolders."
+        extensions = ifnone(extensions, AUDIO_EXTENSIONS)
+        return super().from_folder(path=path, extensions=extensions, **kwargs)
+        
+        
+    @classmethod
+    def from_df(cls, df:DataFrame, path:PathOrStr, cols:IntsOrStrs=0, folder:PathOrStr=None, suffix:str='', **kwargs)->ItemList:
```
Comment (rbracco) [inline audio/data.py:407]: Thanks for adding this, great work!

File: audio/transform.py [hunk 5]
```diff
@@ -266,7 +268,7 @@ def tfm_pad_or_trim(ai:AudioItem, mx, trim_section="mid", pad_at_end=True, **kwa
     return AudioItem(sig=nsig, sr=ai.sr)
 
 def tfm_downmix(signal):
-    return DownmixMono(channels_first=True)(signal)
+    return torch.mean(signal, 0,True)
```
Comment (rbracco) [inline audio/transform.py:263]: looks good

**Review by rbracco — CHANGES_REQUESTED** [review]
Changes look good, I tested the branch locally and everything works. The following info/changes are needed:

1. If you could give a little more info about your decision to change the install process so we can decide if it is the best option
2. General cleanup, removing commented out code and excessive whitespace.

After this it should be good to merge. Thanks again for your contribution and let me know if you have interest in helping to develop the fastai v2 audio module.

File: README.md (hunk 1 above)
Comment (kevinbird15) [reply README.md:11]: My guess is to make it more standard.  I am wondering if an environment.yml file might be better though.  One thing I see here is that I think Jeremy at least would prefer to use the pip version of librosa over the conda-forge version.  I could create the environment.yml file, but I'm not sure if there is a way to create a 'tack-on' environment.yml file where it won't create a whole new environment, but will just add the libraries that you want to include.

Comment (filipmu) [reply README.md:11]: The reason I changed it was that pytorch, torchaudio, and fastai are installed via conda and the conda docs recommend avoiding installing packages via pip if conda was used.  See https://www.anaconda.com/using-pip-in-a-conda-environment/

Also, based on my install  fastai_audio only needed  a conda install of librosa and then it ran fine through all the tutorial workbooks.  I am not sure it uses pydub or fire libraries.  Also I think conda torchaudio installs the sox dependency for torchaudio automatically.  Worth a second set of eyes tho.

File: audio/transform.py (hunk 3 above)
Comment (filipmu) [reply audio/transform.py]: I made this change.

File: audio/data.py (hunk 2 above)
Comment (filipmu) [reply audio/data.py]: This is now complete.

File: README.md (hunk 1 above)
Comment (mogwai) [reply README.md:11]: For now I think just keep the install script. It's an easy way for people to get up and running and documents what the requirements are. I agree @kevinbird15  that a envrionment.yml would be better but we're not planning on supporting this version of the library when v2 is complete. 

Add the install.sh back into the readme and change the install script to install torchaudio==0.3.1 on the last line and we can merge this PR.

Great job @filipmu thank you

**Comment by mogwai** [comment]
Still waiting for changes to the install.sh script and adding @whoahbot suggestions.

**Pull request merged**